src/Core/Utils/Index.hpp

```diff
@@ -10,10 +10,12 @@ namespace Utils {
 
 class RA_CORE_API Index
 {
+    using IntegerType = int;
+
   public:
```

Comment:
```suggestion
  public:
    // Alias type used to store integers
    using IntegerType = int;
    
```
We want this type public as it is used as input of a public constructor